src/bin/solve.rs

```diff
@@ -51,14 +52,68 @@ fn generate_fwopcache<P: AsRef<Path>>(path: P) -> io::Result<File> {
             })
             .or_insert_with(|| (i, i.count_ones()));
     }
-    for (k, v) in cache.entries() {
-        file.write(k.to_ne_bytes())?;
-        file.write(&v.to_ne_bytes())?;
+
+    for (k, v) in &cache {
+        file.write(&k.to_ne_bytes())?;
+        file.write(&v.0.to_ne_bytes())?;
     }
     file.flush()?;
     Ok(file)
 }
 
+// fn get_entry_from_cache<F: Read, F:Seek>(cache: F) -> Option<u64> {
+
+// }
+
+// fn write_cache<F: Write>(cache: &BTreeMap<u32, (u32, u32)>, buf: &mut F) {
+
+// }
+
+fn get_entry_from_cache(cache: &mut File, x: u32) -> Option<u32> {
+    let mut base = 0;
+    let mut size = 1 << 23; // Num entries in cache
+    while size > 0 {
+        size /= 2;
+        let mid = (base + size) / 2;
+        // get this entry from cache
+        let mut buf = [0; 4];
+        cache.seek(io::SeekFrom::Start(mid as u64 * 8)).unwrap();
+        cache
+            .read_exact(&mut buf)
+            .map_err(|_| {
+                io::Error::new(
+                    io::ErrorKind::UnexpectedEof,
+                    "Failed to find a solution in the cache",
+                )
+            })
+            .unwrap();
```

Comment:
Haha what the heck, you're doing `map_err` and then `unwrap`, throwing away the error formatting. Change the return type to an `io::Result<u32>` and use `?` instead.